usaspending_api/database_scripts/job_archive/backfill_per_transaction_exec_comp.py

```diff
@@ -0,0 +1,322 @@
+    return (_max - min_id + 1) / total
+
+
+def print_no_rows_to_update(transaction_type):
+    progress = determine_progress()
+    print(
+        "[{} - {:.2%}] {:,} => {:,}: No rows to update with an estimated remaining run time of {}"
+        .format(
+            transaction_type, progress, _min, _max, chunk_timer.estimated_remaining_runtime(progress)
+        ),
+        flush=True
+    )
+
+
+def run_broker_select_query(transaction_sql):
+    with broker_connection.cursor() as select_cursor:
+        query_parameters = {
+            "min_id": _min,
+            "max_id": _max
+        }
+        select_cursor.execute(transaction_sql, query_parameters)
+        return select_cursor.fetchall()
+
+
+def run_spending_update_query(transaction_sql, transaction_type, broker_data):
+    with spending_connection.cursor() as update_cursor:
+        update_query = build_spending_update_query(transaction_sql, broker_data)
+        with Timer() as t:
+            update_cursor.execute(
+                update_query,
+                [col for row in broker_data for col in row]
+            )
+        row_count = update_cursor.rowcount
+        progress = determine_progress()
+        print(
+            "[{} - {:.2%}] {:,} => {:,}: {:,} rows updated in {} with an estimated remaining run time of {}"
+            .format(
+                transaction_type, progress, _min, _max, row_count, t.elapsed_as_string,
+                chunk_timer.estimated_remaining_runtime(progress)
+            ),
+            flush=True
+        )
+        return row_count
+
+
+with Timer() as overall_timer:
+    with psycopg2.connect(dsn=SPENDING_CONNECTION_STRING) as spending_connection, \
+            psycopg2.connect(dsn=BROKER_CONNECTION_STRING) as broker_connection:
+        spending_connection.autocommit = True
+
+        print("Running FABS backfill from Broker to USAspending")
+
+        fabs_row_count = 0
+
+        with broker_connection.cursor() as cursor:
+            print("Finding min/max Published_Award_Financial_Assistance_ID for FABS...")
+            cursor.execute(GET_MIN_MAX_FABS_SQL)
+            results = cursor.fetchall()
+            min_id, max_id = results[0]
+            total = max_id - min_id + 1
+
+        print("Min Published_Award_Financial_Assistance_ID: {:,}".format(min_id))
+        print("Max Published_Award_Financial_Assistance_ID: {:,}".format(max_id), flush=True)
+
+        with Timer() as chunk_timer:
+            _min = min_id
+            while _min <= max_id:
+                _max = min(_min + CHUNK_SIZE - 1, max_id)
+                broker_fabs_data = run_broker_select_query(BROKER_FABS_SELECT_SQL)
+                if broker_fabs_data:
+                    updated_row_count = run_spending_update_query(SPENDING_FABS_UPDATE_SQL, "FABS", broker_fabs_data)
+                    fabs_row_count += updated_row_count
+                else:
+                    print_no_rows_to_update("FABS")
+                _min = _max + 1
```

Comment:
```suggestion
               # move to next chunk
                _min = _max + 1
```
Not important, just an idea

Reply:
I can make this change to improve readability and then go the route of Kirk's abstracted _min and _max id generator in any future throwaway scripts.